vowpalwabbit/parse_example.cc

```diff
@@ -50,29 +50,113 @@ int read_features_string(vw* all, v_array<example*>& examples)
   return static_cast<int>(num_chars_initial);
 }
 
+class feature_group_helper
+{
+public:
+  using char_map = std::array<unsigned char, NUM_NAMESPACES>;
+  using affix_map = std::array<uint64_t, NUM_NAMESPACES>;
+  using bool_map = std::array<bool, NUM_NAMESPACES>;
+  using namespace_dictionaries = std::vector<std::shared_ptr<feature_dict>>;
+  using dictionaries_map = std::array<namespace_dictionaries, NUM_NAMESPACES>;
+
+  feature_group_helper(example* current_example, bool redefine_char_flag, char_map* redefine_char_map,
+      affix_map* affix_features, bool_map* spelling_features, dictionaries_map* namespace_dictionaries_map,
+      uint32_t hash_seed, hash_func_t hash_fn)
+      : _feature_group_char(0)
+      , _hash(0)
+      , _hash_seed(hash_seed)
+      , _p_current_feature_group(nullptr)
+      , _p_affix_feature_group(nullptr)
+      , _p_spelling_feature_group(nullptr)
+      , _p_dictionary_feature_group(nullptr)
+      , _current_example(current_example)
+      , _redefine_char_flag(redefine_char_flag)
+      , _redefine_char_map(redefine_char_map)
+      , _namespace_dictionaries_map(namespace_dictionaries_map)
+      , _affix_features(affix_features)
+      , _spelling_features(spelling_features)
+      , _hash_function(hash_fn)
+  {
+  }
+  inline unsigned char feature_group_char() const { return _feature_group_char; }
```

Comment:
aren't member function inherently inline in C++?

Reply:
I see member functions that have function definitions within class definitions are implicitly inlined.  Ok.  Will change this.

Reply:
done.